view/statistics-flow-certificates.js

```diff
@@ -0,0 +1,312 @@
+  , Pagination        = require('./components/pagination')
+  , modes             = require('../utils/statistics-flow-group-modes')
+  , selectService     = require('./components/filter-bar/select-service')
+  , selectCertificate = require('./components/filter-bar/select-certificate')
+  , selectPeriodMode  = require('./components/filter-bar/select-period-mode')
+  , itemsPerPage      = require('../conf/objects-list-items-per-page')
+  , isNaturalNumber   = require('es5-ext/number/is-natural');
+
+exports._parent        = require('./statistics-flow');
+exports._customFilters = Function.prototype;
+
+exports['flow-nav']                = { class: { 'submitted-menu-item-active': true } };
+exports['flow-by-certificate-nav'] = { class: { 'pills-nav-active': true } };
+
+var queryServer = memoize(function (query) {
+	return getData('/get-flow-data/', query);
+}, {
+	normalizer: function (args) { return JSON.stringify(args[0]); },
+	max: 1000
+});
+
+var serviceToCertLegacyMatch = { '': [] };
+
+var getServiceName = function (ServiceType) {
+	return uncapitalize.call(
+		ServiceType.__id__.slice('BusinessProcess'.length)
+	);
+};
+
+db.BusinessProcess.extensions.forEach(function (ServiceType) {
+	ServiceType.prototype.certificates.map.forEach(function (certificate) {
+		if (!serviceToCertLegacyMatch[getServiceName(ServiceType)]) {
+			serviceToCertLegacyMatch[getServiceName(ServiceType)] = [];
+		}
+		serviceToCertLegacyMatch[getServiceName(ServiceType)].push('certificate-' + certificate.key);
+		serviceToCertLegacyMatch[''].push('certificate-' + certificate.key);
+	});
+});
+
+var buildResultRow = function (rowData) {
+	return assign({
+		submitted: 0,
+		pending: 0,
+		pickup: 0,
+		withdrawn: 0,
+		rejected: 0,
+		sentBack: 0
+	}, rowData || {});
+};
+
+var accumulateResultRows = function (rows) {
+	var result = buildResultRow(rows[0]);
+	rows.slice(1).forEach(function (row) {
+		Object.keys(row).forEach(function (propertyName) {
+			result[propertyName] += row[propertyName];
+		});
+	});
+
+	return result;
+};
+
+var filterData = function (data) {
+	var result = {};
+	location.query.get('service').map(function (service) {
+		location.query.get('certificate').map(function (certificate) {
+			Object.keys(data).forEach(function (key) {
+				if (service && certificate) {
+					result[key] = buildResultRow(data[key][service].certificate[certificate]);
+				} else if (service) {
+					result[key] = buildResultRow(data[key][service].businessProcess);
+				} else {
+					var rowsToAccumulate = [];
+					Object.keys(data[key]).forEach(function (service) {
+						if (certificate) {
+							if (!data[key][service].certificate[certificate]) return;
+							rowsToAccumulate.push(data[key][service].certificate[certificate]);
+						} else {
+							rowsToAccumulate.push(data[key][service].businessProcess);
+						}
+					});
+					result[key] = accumulateResultRows(rowsToAccumulate);
+				}
+			});
+		});
```

Comment:
I take that server can send data with holes e.g. omit days for which it doesn't have data. So result can be:

```javascript
{
  '2016-01-23': ...
  '2016-01-27': ...
  '2016-02-01': ...
}
```

It would also be optimal for size of request.

In such case I believe we should fill missing dates with `buildResultRow()`